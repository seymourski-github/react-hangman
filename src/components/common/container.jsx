import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const getStyles = props => {
  const { flexDirection, flexItem, flexWrap, style } = props;
  if(!style && !flexItem) return;
  return {
    ...style,
    flex: flexItem,
    flexDirection: flexDirection,
    flexWrap: flexWrap
  };
};

const getClass = props => {
  const { flex } = props;
  return classNames('container', {
    column: flex && flex==='column',
    row: flex===true || flex==='row',
  });
};

export const Container = props => {
  const { id, children } = props;
  if(!children) return null;
  return (
    <div className={getClass(props)} id={id} style={getStyles(props)}>
      {children}
    </div>
  );
};

Container.propTypes = {
  flex: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string
  ]),
  flexDirection: PropTypes.string,
  flexWrap: PropTypes.string,
  style: PropTypes.object,
  id: PropTypes.string
};

Container.defaultProps = {
  className: 'container',
  id: null,
  flex: 'row',
  flexDirection: 'row',
  flexWrap: 'wrap',
  style: null
};
