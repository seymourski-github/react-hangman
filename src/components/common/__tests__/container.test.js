import React from 'react';
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { Container } from '../container';

const props = {
  id: "my-container"
};

describe('<Container /> Component', function() {

  const component = renderer.create(<Container {...props} />);
  let wrapper = shallow(<Container {...props} />);

  it("renders and matches our snapshot", () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
