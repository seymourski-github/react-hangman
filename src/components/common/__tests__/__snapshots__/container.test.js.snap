// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> Component renders and matches our snapshot 1`] = `null`;
